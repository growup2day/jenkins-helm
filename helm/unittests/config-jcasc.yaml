templates:
  - config-jcasc.yaml
tests:
  - it: default JCasC should have no data
    asserts:
      - isNull:
          path: data
  - it: should process custom configScripts
    set:
      JCasC:
        configScripts:
          jenkins-global: |
            jenkins:
              systemMessage: This Jenkins is configured and managed 'as code'.
              mode: EXCLUSIVE
              numExecutors: 0
          init-scripts: |
            groovy:
              - script: >
                  println("This is Groovy script!");
    asserts:
      - matchSnapshot:
          path: data
  - it: should generate blank default config script when enabled
    set:
      domain: dummy.domain
      JCasC:
        defaultConfig: true
    asserts:
      - matchSnapshot:
          path: data
