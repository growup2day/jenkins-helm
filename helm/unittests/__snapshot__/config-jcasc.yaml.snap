should generate blank default config script when enabled:
  1: |
    jcasc-default-config-basic.yaml: |-
      jenkins:
        # Configure System > Maven Project Configuration
        mode: "NORMAL"
        numExecutors: 0
      unclassified:
        # Configure System > Jenkins Location
        location:
          url: "https://RELEASE-NAME-jenkins-NAMESPACE.dummy.domain/"
          adminAddress: ""
    jcasc-default-config-bitbucket.yaml: |-
      unclassified:
        # Configure System > Bitbucket Endpoints
        bitbucketEndpointConfiguration:
          endpoints: []
    jcasc-default-config-clouds.yaml: |-
      jenkins:
        # Manage Nodes and Clouds > Configure Clouds
        clouds: []
    jcasc-default-config-credentials.yaml: |-
      # Manage Credentials
      credentials:
        system:
          domainCredentials:
            - credentials:
                - vaultKubernetesCredential:
                    description: "Vault Auth"
                    id: "vault-auth"
                    mountPath: "kubernetes"
                    role: "microservice"
                    scope: "SYSTEM"
    jcasc-default-config-env-vars.yaml: |-
      jenkins:
        # Configure System > Global properties
        globalNodeProperties:
          - envVars:
              env: []
    jcasc-default-config-libraries.yaml: |-
      unclassified:
        # Configure System > Global Pipeline Libraries
        globalLibraries:
          libraries: []
    jcasc-default-config-mailer.yaml: |-
      unclassified:
        # Configure System > E-mail Notification
        mailer:
          smtpHost: ""
    jcasc-default-config-nodes.yaml: |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes: []
    jcasc-default-config-openshift-sync.yaml: |-
      unclassified:
        # Configure System > OpenShift Jenkins Sync
        globalPluginConfiguration:
          enabled: false
    jcasc-default-config-openshift.yaml: |-
      unclassified:
        # Configure System > OpenShift Client Plugin
        openShift:
          clusterConfigs: []
    jcasc-default-config-proxy.yaml: |-
      jenkins:
        # Manage Plugins > Advanced
        proxy:
    jcasc-default-config-script-approval.yaml: |-
      security:
        # In-process Script Approval
        scriptApproval:
          approvedSignatures: []
    jcasc-default-config-slack.yaml: |-
      unclassified:
        # Configure System > Slack
        slackNotifier:
          teamDomain: ""
          tokenCredentialId: ""
    jcasc-default-config-vault.yaml: |-
      unclassified:
        # Configure System > Vault Plugin
        hashicorpVault:
          configuration:
            vaultCredentialId: ""
            vaultNamespace: ""
            vaultUrl: ""
should process custom configScripts:
  1: |
    init-scripts.yaml: |-
      groovy:
        - script: >
            println("This is Groovy script!");
    jenkins-global.yaml: |-
      jenkins:
        systemMessage: This Jenkins is configured and managed 'as code'.
        mode: EXCLUSIVE
        numExecutors: 0
